functions.py

```diff
@@ -0,0 +1,189 @@
+''' Shared library of functions for report.py and remind.py
+'''
+
+import os
+import re
+import datetime
+import bugzilla
+from jira import JIRA
+
+
+def generate_html_file(htmlcode, remind=False):
+	''' generates HTML file of reminder
+	'''
+	try:
+		os.makedirs('archive')
+	except FileExistsError:
+		pass
+	reportType = 'reminder' if remind else 'report'
+	filename = './archive/{}_{:%m%d%Y_%H:%M:%S}.html'.format(
+		reportType, datetime.datetime.now())
+	with open(filename, 'w') as file:
+		file.write(htmlcode)
+	return None
+
+
+def get_bugs_dict(bug_ids, config):
+	''' takes in set of bug_ids and returns dictionary with
+		bug_ids as keys and API data as values
+	'''
+
+	# initialize bug dictionary
+	bugs = {}
+
+	# iterate through bug ids from set
+	for bug_id in bug_ids:
+
+		# 0 should be default in YAML file (i.e. no bugs recorded)
+		# if present reference should be made in bug dict
+		if bug_id == 0:
+			bugs[0] = {'bug_name': 'No bug on file', 'bug_url': None}
+			continue
+
+		# get bug info from bugzilla API
+		try:
+
+			# hotfix: API call does not work if '/' present at end of URL string
+			parsed_bz_url = config['bz_url'].rstrip('/')
+
+			bz_api = bugzilla.Bugzilla(parsed_bz_url)
+			bug = bz_api.getbug(bug_id)
+			bug_name = bug.summary
+		except Exception as e:
+			print("Bugzilla API Call Error: ", e)
+			bug_name = "BZ#" + str(bug_id)
+		finally:
+			bug_url = config['bz_url'] + "/show_bug.cgi?id=" + str(bug_id)
+			bugs[bug_id] = {'bug_name': bug_name, 'bug_url': bug_url}
+
+	return bugs
+
+
+def get_bugs_set(blockers):
+	''' takes in blockers object and generates a set of all unique bug ids
+		including 0 if it is present
+	'''
+	bug_set = {}
```

Comment:
Did you mean this to be a ``set()``? Can't ``add`` to a dict..

Reply:
I recommend using the previous logic to keep this as a list and use extend on line 69 to add the list of BZs to the list..

Reply:
Right, that's what the previous iteration did. Going to revert to that; can be changed in a future patch.